File: frontend/src/app/models/dto/appointment-history-dto.ts
import dayjs from 'dayjs';
import { Reason } from './reason.enum';
import { Speciality } from './speciality.enum';
import { Status } from './status.enum';

export interface AppointmentHistoryDTO {
  appointmentId: number;
  startTime: string;
  endTime: string;
  doctorName: string;
  doctorId: number;
  doctorSpeciality: Speciality;
  clientName: string;
  petName: string;
  petId: number;
  reason: Reason;
  status: Status;
  hasDiagnosis: boolean;
  diagnosisId: number | null;
}

export function mapAppointmentDateToDate(dto: AppointmentHistoryDTO) {
  const start = dto.startTime ? dayjs(dto.startTime).local() : null;
  const end = dto.endTime ? dayjs(dto.endTime).local() : null;

  return {
    ...dto,
    startDate: start?.isValid() ? start.toDate() : null,
    endDate: end?.isValid() ? end.toDate() : null,
  } as AppointmentHistoryDTO & { startDate: Date | null; endDate: Date | null };
}

File: backend/Pet-Society/src/main/java/Pet/Society/models/dto/appointment/AppointmentHistoryDTO.java
package Pet.Society.models.dto.appointment;

import Pet.Society.models.enums.Reason;
import Pet.Society.models.enums.Speciality;
import Pet.Society.models.enums.Status;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Data
@SuperBuilder
public class AppointmentHistoryDTO {
    private Long appointmentId;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private String doctorName;
    private Long doctorId;
    private Speciality doctorSpeciality;
    private String clientName;
    private String petName;
    private Long petId;
    private Reason reason;
    private Status status;
    private boolean hasDiagnosis;
    private Long diagnosisId; // Para poder obtener el diagnóstico después si es necesario
}

